spyder/plugins/projects/plugin.py

```diff
@@ -409,6 +433,47 @@ def get_project_types(self):
         """
         return self.get_widget().get_project_types()
 
+    def handle_switcher_modes(self, mode):
+        """
+        Populate switcher with files in active project.
+        List the file names of the current active project with their
+        directories in the switcher. Only handle file mode, where
+        `mode` is empty string.
+        Parameters
+        ----------
+        mode: str
+            The selected mode (open files "", symbol "@" or line ":").
+        """
+        self.get_widget().handle_switcher_modes("")
+
+    def handle_switcher_selection(self, item, mode, search_text):
+        """
+        Handle user selecting item in switcher.
+        If the selected item is not in the section of the switcher that
+        corresponds to this plugin, then ignore it. Otherwise, switch to
+        selected item in notebook plugin and hide the switcher.
```

Comment:
Similiar suggestion for the docstring, Maybe here the docstring should be `project file` instead of `item in notebook plugin`. So something like:

```suggestion
        selected project file and hide the switcher.
```